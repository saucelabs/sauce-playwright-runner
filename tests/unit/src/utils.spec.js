const utils = require('../../../src/utils');

describe('utils', function () {
  describe('.logHelper', function () {
    let nowSpy;
    let fakeTime = 0;
    beforeEach(function () {
      global.logs = [];
      global._timeOfLastCommand = 0;
      nowSpy = jest.spyOn(Date, 'now');
      nowSpy.mockImplementation(() => fakeTime += 1000);
    });
    afterEach(function () {
      nowSpy.mockClear();
    });
    it('should add to global logs', function () {
      const logMessage = {
        params: 'some-param',
        id: 'foo',
        method: 'warn',
      };
      utils.logHelper(null, null, `SEND â–º ${JSON.stringify(logMessage)}`);
      expect(global.logs).toMatchSnapshot();
    });
  });
});