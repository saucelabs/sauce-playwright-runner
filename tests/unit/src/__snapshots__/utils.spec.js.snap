// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils .logHelper should add to global logs 1`] = `
Array [
  Object {
    "HTTPStatus": 200,
    "between_commands": 1,
    "duration": 1,
    "id": "foo",
    "in_video_timeline": 0,
    "method": "SOCKET",
    "path": "warn",
    "request": "some-param",
    "result": "",
    "screenshot": 0,
    "start_time": 2,
    "statusCode": 0,
    "suggestion": null,
    "suggestion_values": Array [],
  },
]
`;
